helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1925,7 +1931,10 @@ private void validate() {
 
       // Resolve RealmAwareZkClientConfig
       if (realmAwareZkClientConfig == null) {
-        realmAwareZkClientConfig = new RealmAwareZkClient.RealmAwareZkClientConfig();
+        // ZkHelixAdmin should have ZNRecordSerializer set by default
+        realmAwareZkClientConfig = new RealmAwareZkClient.RealmAwareZkClientConfig()
+            .setZkSerializer(
+                new org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer());
```

Comment:
Nit, I think currently we can still use the deprecated ZNRecordSerializer. Once the deprecated one is removed, we will also refactor and change it here. If we put the long path here, maybe we would not find it when removing the deprecated one.

Reply:
But then we'd be using a deprecated component. I don't think using the package prefix is wrong per se..